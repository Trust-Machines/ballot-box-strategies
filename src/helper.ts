import {StacksMainnet, StacksTestnet} from '@stacks/network';
import {NETWORK} from './types';
import strategies, {runStrategy} from './index';

export const getNetworkByName = (name: NETWORK) => {
    switch (name) {
        case 'mainnet':
            return new StacksMainnet();
        case 'testnet':
            return new StacksTestnet();
        default:
            throw new Error('Unknown network!');
    }
}

export const testStrategy = async (s: string) => {
    const strategy = strategies[s];
    const {testConfig} = strategy;

    for (let address of testConfig.addresses) {
        const resp = await runStrategy(s, getNetworkByName(testConfig.network), address, testConfig.blockTip, testConfig.options);
        expect(resp).toMatchSnapshot(`${s}-${address}`);
    }
}
