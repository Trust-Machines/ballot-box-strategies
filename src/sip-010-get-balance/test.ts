import {testStrategy} from '../helper';

it('sip-010-get-balance', async () => {
    await testStrategy('sip-010-get-balance');
});
