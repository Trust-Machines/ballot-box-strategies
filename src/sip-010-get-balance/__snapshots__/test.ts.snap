// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sip-010-get-balance: sip-010-get-balance-SP2DXHX9Q844EBT80DYJXFWXJKCJ5FFAX50CQQAWN 1`] = `0.1`;

exports[`sip-010-get-balance: sip-010-get-balance-SP2N7SK0W83NJSZHFH8HH31ZT3DXJG7NFE5VYT9SJ 1`] = `0.7405`;

exports[`sip-010-get-balance: sip-010-get-balance-SP2VWSP59FEVDXXYGGWYG90M3N67ZST2AGQK6Q5RY 1`] = `0`;

exports[`sip-010-get-balance: sip-010-get-balance-SP3XD84X3PE79SHJAZCDW1V5E9EA8JSKRBPEKAEK7 1`] = `2.7972`;
